### ISSUE TITLE
SubModelSeq deselects items when moving

### ISSUE THREAD

**Issue by cmeeren** [elmish/Elmish.WPF#302]
In the `SubModelSeq` sample. when moving counters up/down, they are deselected. I don't think that happened before. Unless I'm mistaken, it would be good if we could bring back that behavior.

(This is not critical and not something I will prioritize looking into myself.)

**Comment by TysonMN** [comment]
I can fix this

**Comment by TysonMN** [comment]
I think the proper way to do this is to store the selected item in the model.  That is not currently happening.

Specifically looking at the `TreeView` in the head of `master`...
https://github.com/elmish/Elmish.WPF/blob/8e8d3b00f6ec728943a51497f5af17348963fe07/src/Samples/SubModelSeq.Views/MainWindow.xaml#L23

...and a bit of git history for the corresponding file, I don't think there ever was a binding to `SelectedItem`.  Maybe a previous version of the code maintained the selected item by somewhat coincidently storing the selected item in WPF.

**Comment by cmeeren** [comment]
> Maybe a previous version of the code maintained the selected item by somewhat coincidently storing the selected item in WPF.

Yes, I think that's correct. It just worked. I'm curious why that is not the case. Are you able and willing to investigate that? It's not critical, but if we could restore that behavior, so that we don't have to "bog down" the app logic with managing the selected item, I think that would be best.

**Comment by TysonMN** [comment]
> Are you able and willing to investigate that?

Definitely willing.  I will find the time, but not right away.

**Comment by BentTranberg** [comment]
Has something happened going from .NET Framework to .NET Core / .NET 5 ? Regression or intended changes? Is this related to #440 ? I have had a feeling several times that there are subtle changes in this area in WPF.

**Comment by the-not-mad-psychologist** [comment]
In v3.5.5 and older, selection is kept when moving up and down, except when the selected item is at the top or bottom of the list and another is moved into its position, in which case the replacing item receives selection.
